### ISSUE TITLE
Random Seed Parameter

### ISSUE THREAD

**Issue by dwrout** [ollama/ollama#2773]
Currently, the Ollama  "seed" parameter is set as a fixed preset number that remains constant throughout an interaction unless changed manually by the user or API. I propose an enhancement allowing this seed parameter to be random as a preset option. This could add more variability and unpredictability to the chat responses. Because the seed parameter is a positive integer, using a negative number could invoke the random function. 

Here are some thoughts on implementing this feature:
1. Allow users to set "-1" as a seed parameter so that each new chat started uses a new random number as the seed parameter for the duration of the session.
2. Allow users to set "-2" as a seed parameter so that each new chat response within a chat session uses a new random number as the seed parameter.
3. Allow users to set "-3" as a seed parameter to generate and automatically use unique seeds based on specific conditions, such as asking the same question again.

**Comment by Crytoma** [comment]
Hear hear!

**Comment by 9cento** [comment]
+1

**Comment by jmorganca** [comment]
Hi folks, this should be fixed in https://github.com/ollama/ollama/pull/4986. Thanks so much for creating an issue.

RE 1-3 above, you should be able to do this via the `seed` parameter sent with each request – generally the Ollama API tries to be "stateless" and the client can send the `seed`. Let me know if that helps!

**Comment by cirosantilli** [comment]
> Hi folks, this should be fixed in [#4986](https://github.com/ollama/ollama/pull/4986). Thanks so much for creating an issue.
> 
> RE 1-3 above, you should be able to do this via the `seed` parameter sent with each request – generally the Ollama API tries to be "stateless" and the client can send the `seed`. Let me know if that helps!

How to set the seed from CLI? I had a look at: https://github.com/ollama/ollama/issues/2505 and naively tried:

```
/bin/time ollama run llama2 '/set parameter seed 0' 'What is quantum field theory?'
```

but it didn't seem to work.

**Comment by cirosantilli** [comment]
Doing:

```
ollama run llama3.2
```

and then manually on terminal:

```
>>> /set parameter seed 0
Set parameter 'seed' to '0'
>>> What is quantum field theory?
```

was also not deterministic.
